### PR TITLE
Allow removal of .mem files alongside .cov files

### PR DESCRIPTION
**Pull request by bluesmoon**

### CODE REVIEW

**Comment by bluesmoon** [comment]
I don't understand this test failure.  It looks more like a change in the way Julia nightly reports a particular exception:
```
Processing coverage: Test Failed at /home/runner/work/CoverageTools.jl/CoverageTools.jl/test/runtests.jl:200
  Expression: process_file(bustedfile, srcdir)
    Expected: Base.Meta.ParseError("parsing error in /home/runner/work/CoverageTools.jl/CoverageTools.jl/test/BustedPackage/src/parseerr.jl:7: invalid iteration specification", nothing)
      Thrown: Base.Meta.ParseError("parsing error in /home/runner/work/CoverageTools.jl/CoverageTools.jl/test/BustedPackage/src/parseerr.jl:8: Base.Meta.ParseError(\"ParseError:\\n# Error @ none:3:10\\n    s = 0\\n    for i [1,2,3]   # this line has a parsing error\\n#        └─┘ ── invalid iteration spec: expected one of `=` `in` or `∈`\", Base.JuliaSyntax.ParseError(Base.JuliaSyntax.SourceFile(\"function parseerr()\\n    s = 0\\n    for i [1,2,3]   # this line has a parsing error\\n        \", 42, \"none\", 1, [1, 21, 31, 83, 91]), Base.JuliaSyntax.Diagnostic[Base.JuliaSyntax.Diagnostic([82](https://github.com/JuliaCI/CoverageTools.jl/actions/runs/6554881300/job/17802524253?pr=72#step:5:85), 84, :error, \"invalid iteration spec: expected one of `=` `in` or `∈`\"), Base.JuliaSyntax.Diagnostic(87, 86, :error, \"invalid iteration spec: expected one of `=` `in` or `∈`\"), Base.JuliaSyntax.Diagnostic(89, 88, :error, \"invalid iteration spec: expected one of `=` `in` or `∈`\"), Base.JuliaSyntax.Diagnostic(89, 89, :error, \"unexpected `]`\"), Base.JuliaSyntax.Diagnostic(89, 88, :error, \"Expected `end`\"), Base.JuliaSyntax.Diagnostic(89, 88, :error, \"Expected `end`\"), Base.JuliaSyntax.Diagnostic(89, 89, :error, \"extra tokens after end of expression\")], :none))", nothing)
      ParseError("parsing error in /home/runner/work/CoverageTools.jl/CoverageTools.jl/test/BustedPackage/src/parseerr.jl:8: Base.Meta.ParseError(\"ParseError:\\n# Error @ none:3:10\\n    s = 0\\n    for i [1,2,3]   # this line has a parsing error\\n#        └─┘ ── invalid iteration spec: expected one of `=` `in` or `∈`\", Base.JuliaSyntax.ParseError(Base.JuliaSyntax.SourceFile(\"function parseerr()\\n    s = 0\\n    for i [1,2,3]   # this line has a parsing error\\n        \", 42, \"none\", 1, [1, 21, 31, [83](https://github.com/JuliaCI/CoverageTools.jl/actions/runs/6554881300/job/17802524253?pr=72#step:5:86), 91]), Base.JuliaSyntax.Diagnostic[Base.JuliaSyntax.Diagnostic(82, [84](https://github.com/JuliaCI/CoverageTools.jl/actions/runs/6554881300/job/17802524253?pr=72#step:5:87), :error, \"invalid iteration spec: expected one of `=` `in` or `∈`\"), Base.JuliaSyntax.Diagnostic(87, 86, :error, \"invalid iteration spec: expected one of `=` `in` or `∈`\"), Base.JuliaSyntax.Diagnostic(89, 88, :error, \"invalid iteration spec: expected one of `=` `in` or `∈`\"), Base.JuliaSyntax.Diagnostic(89, 89, :error, \"unexpected `]`\"), Base.JuliaSyntax.Diagnostic(89, 88, :error, \"Expected `end`\"), Base.JuliaSyntax.Diagnostic(89, 88, :error, \"Expected `end`\"), Base.JuliaSyntax.Diagnostic(89, 89, :error, \"extra tokens after end of expression\")], :none))")
      Stacktrace:
```

**Comment by bluesmoon** [comment]
bump - can anyone respond here?

**Comment by Azzaare** [comment]
Ran into the same issue with #73 

It comes from the Julia parser, giving different output with each Julia version. I have fixed it for current `1` (`1.10`). If my PR is accepted, you can probably rebase this PR over main and have your tests pass.

**Comment by bluesmoon** [comment]
@Azzaare I doubt anything will be accepted here. The last activity on this repo was 3 years ago.

**Comment by Azzaare** [comment]
@bluesmoon PR #73 was merged! You can probably rebase on master now and have the tests pass!

File: src/CoverageTools.jl [hunk 1]
```diff
@@ -267,51 +267,60 @@ function process_folder(folder="src")
...
 # with directories, but the directories must match
 function iscovfile(filename, sourcefile)
     startswith(filename, sourcefile) || return false
-    occursin(r"\.jl\.?[0-9]*\.cov$", filename)
+    iscovfile(filename)
 end
 
 """
-    clean_folder(folder::AbstractString)
+    clean_folder(folder::AbstractString; include_memfiles::Bool=false)
```
Comment (fingolfin) [inline src/CoverageTools.jl:289]: Maybe `include_memfiles` is not the best name (`also_memfiles`? also not great)

Well naming is hard. I think it is good enough 🤷

**Review by fingolfin — APPROVED** [review]

**Pull request merged**